typify-impl/src/convert.rs

```diff
@@ -1163,6 +1163,120 @@ impl TypeSpace {
                 )?;
                 Ok((type_entry, metadata))
             }
+
+            Some(ObjectValidation {
+                max_properties: _,
+                min_properties: _,
+                required,
+                properties,
+                pattern_properties,
+                additional_properties,
+                property_names,
+            }) if required.is_empty()
+                && properties.is_empty()
+                // If no pattern properties are provided
+                && (!pattern_properties.is_empty()
+                    && pattern_properties
+                        .values()
+                        .next()
+                        .map(|first| pattern_properties.values().all(|schema| schema == first))
+                        // Default case true because we already checked that pattern_properties is not empty
+                        .unwrap_or(true)
+                    && (property_names.is_none()
+                        || property_names.as_ref().is_some_and(|property_names| {
+                            let Schema::Object(property_names) = property_names.as_ref() else {
+                                return false;
+                            };
+
+                            property_names
+                                .string
+                                .as_ref()
+                                .map(|string| string.pattern.is_none())
+                                .unwrap_or(true)
+                        }))
+                    && additional_properties.as_ref().map(AsRef::as_ref).is_some_and(|schema| {
+                            // This is OK if additional properties is empty, is equal to the pattern properties schema, or is a boolean
+                            matches!(schema, Schema::Bool(_))
+                            || pattern_properties.values().next().is_some_and(|first| first == schema)
+                    })
```

Comment:
this predicate is much too complicated; if we need this, let's move it to a function

Reply:
Narrowing the scope of what is being handled simplified this quite a bit. If it's still too complex, I can move it.